gems/aws-sdk-core/lib/aws-sdk-core/shared_config.rb

```diff
@@ -134,27 +122,28 @@ def assume_role_web_identity_credentials_from_config(profile)
       if @config_enabled && @parsed_config
         entry = @parsed_config.fetch(p, {})
         if entry['web_identity_token_file'] &&
-          entry['role_arn']
+           entry['role_arn']
           AssumeRoleWebIdentityCredentials.new(
             role_arn: entry['role_arn'],
             web_identity_token_file: entry['web_identity_token_file'],
             role_session_name: entry['role_session_name']
           )
-        else
-          nil
         end
-      else
-        nil
       end
     end
 
-    def endpoint_discovery(opts={})
-      get_config_value('endpoint_discovery_enabled', opts)
+    # Add an accessor method (similar to attr_reader) to return a configuration value
+    # Uses the get_config_value below to control where
+    # values are loaded from
+    def self.config_reader(*attrs)
+      attrs.each do |attr|
+        define_method(attr) { |opts = {}| get_config_value(attr.to_s, opts) }
+      end
     end
 
-    def credentials_process(profile)
-      get_config_value('credential_process', profile: profile)
-    end
+    config_reader :region, :sts_regional_endpoints, :s3_us_east_1_regional_endpoint,
```

Comment:
Since we add to this somewhat frequently, I'd wrap this method in parenthesis and put each of these on one line. I'd also sort them either alphabetically or group them based on core configurations vs service only (sts, s3), or both.

Reply:
Done - went with alphabetical rather than based on "core" vs service.